### PR TITLE
[Enhancement] Show component ID, council district(s), and length in component form

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/14157

## Testing
**URL to test:** 
https://deploy-preview-1154--atd-moped-main.netlify.app/moped/projects/1939?tab=map

**Steps to test:**

Edit an existing component's attributes. The modal should have some properties visible:

>We have some not-editable component properties that users need to be able to see in UI. They are component ID, council district, and length (in that order).

>The length field should use a thousands separator and be hidden if null.

The properties should not be visible when creating a new component

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js [hunk 1]
```diff
@@ -182,11 +177,10 @@ const ComponentForm = ({
     parentValue: watch("signal"),
     dependentFieldName: "locationDescription",
     comparisonVariable: "properties.id",
-    valueToSet:
-      signal
-      // if the signal exists and the locationDescription is empty, set to option label
-        ? getSignalOptionLabel(signal)
-        : "",
+    valueToSet: signal
```
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:180]: linting

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryCouncilDistricts.js
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryCouncilDistricts.js:1]: linting and renaming the function

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/useProjectComponents.js [hunk 2]
```diff
@@ -1,12 +1,35 @@
 import { useMemo, createRef } from "react";
 
+const setComponentCouncilDistrict = (component, projectGeography) => {
+  const componentID = component.project_component_id;
+  const councilDistricts = projectGeography
+    .filter((f) => f.component_id === componentID)
+    .map((f) => f.council_districts)
+    .flat();
+  component.council_districts = [...new Set(councilDistricts)];
+};
```
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectComponents/utils/useProjectComponents.js:10]: stole this code from Joh'n's original demo of the council districts

**Comment by chiaberry** [comment]
@johnclary which do you like better? 
Screenshot 2023-10-10 at 9 01 05 PM
Screenshot 2023-10-10 at 9 00 21 PM

**Comment by johnclary** [comment]
@chiaberry i love them laid out in-line! 🙌 ✨

**Review by johnclary — CHANGES_REQUESTED** [review]

**Review by johnclary — APPROVED** [review]
It looks beautiful! Thanks for making quick work of this 🙌 🚢 🚢 🚢 🚢 🚢 🚢

**Review by mddilley — APPROVED** [review]
Looks ✨! I was curious where the feature length was going to show in the UI - very cool to see it in place alongside the other properties. 🚢 🚀 🙌

**Pull request merged**